tests/node/test_wakunode_store.nim

```diff
@@ -0,0 +1,662 @@
+
+        # Then the response contains the messages
+        check:
+          queryResponse2.get().messages == archiveMessages[8..<10]
+        
+      asyncTest "Pagination with Excessive Page Size":
+        # Given the first query (1/1)
+        historyQuery.pageSize = 100
+
+        # When making a history query
+        let queryResponse1 = await client.query(historyQuery, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse1.get().messages == archiveMessages[0..<10]
+
+      asyncTest "Pagination with Zero Page Size":
+        # Given the first query (1/1)
+        historyQuery.pageSize = 0
+
+        # When making a history query
+        let queryResponse1 = await client.query(historyQuery, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse1.get().messages == archiveMessages[0..<10]
+
+      asyncTest "Pagination with Mixed Page Size":
+        # Given the first query (1/3)
+        historyQuery.pageSize = 2
+
+        # When making a history query
+        let queryResponse1 = await client.query(historyQuery, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse1.get().messages == archiveMessages[0..<2]
+
+        # Given the next query (2/3)
+        let historyQuery2 = HistoryQuery(
+          cursor: queryResponse1.get().cursor,
+          pubsubTopic: some(pubsubTopic),
+          contentTopics: contentTopicSeq,
+          ascending: true,
+          pageSize: 4
+        )
+
+        # When making the next history query
+        let queryResponse2 = await client.query(historyQuery2, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse2.get().messages == archiveMessages[2..<6]
+
+        # Given the next query (3/3)
+        let historyQuery3 = HistoryQuery(
+          cursor: queryResponse2.get().cursor,
+          pubsubTopic: some(pubsubTopic),
+          contentTopics: contentTopicSeq,
+          ascending: true,
+          pageSize: 6
+        )
+
+        # When making the next history query
+        let queryResponse3 = await client.query(historyQuery3, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse3.get().messages == archiveMessages[6..<10]
+
+      asyncTest "Pagination with Default Page Size (0)":
+        # Given a query with default page size
+        historyQuery = HistoryQuery(
+          pubsubTopic: some(pubsubTopic),
+          contentTopics: contentTopicSeq,
+          ascending: true
+        )
+
+        # When making a history query
+        let queryResponse = await client.query(historyQuery, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse.get().messages == archiveMessages[0..<10]
```

Comment:
Maybe we should change something in the check so if we change the default page size for a number less than 10, that the right number of messages is received and the test passes?

Reply:
Sure, fine by me.

Reply:
Yes! Now that I read again I didn't express well the idea in the comment.
What I mean is that if for some reason we decide to change the default page size, this test can fail because of the hardcoded 10.
Not sure if it's the intention or not

Reply:
Where do you mean "hardcoded 10"?

Reply:
The 10 in `queryResponse.get().messages == archiveMessages[0..<10]`

If for some reason we decide to change in our codebase the default page size for a number less than 10, then this will fail. This test assumes that the default page size is >= 10 and I think we shouldn't make any assumptions about the default size.

Reply:
Scratch all I wrote above, I was assuming the `Default Page Size` was equivalent to `0`. Hanno just answered another thread and I realised what you truly meant. Updating.